alphagov/paas-rubbernecker
# Refactor application

## What

We're refactoring rubbernecker into Golang, as it is more familiar to the team, meaning anyone should be more comfortable applying a fix or a change. By starting fresh, we can also add some tests.

Along with new language, we're bringing some new features:
- Done column, which will pull stories that have been approved between time it's been cached and 5 days before that.
- In memory caching - meaning the API will not get overused by refreshing the page.
- Free members - lists all team members that happen not to be assigned to any story. Currently hidden, due to team setup.
- New layout - the previous one was heavily based on Bootstrap. It hasn't been clean enough and had some issues with "accessibility".

It has some interfaces and support for modules, meaning it should be easier to implement "plugins" and use different systems to Pivotal or PagerDuty.

To be reviewed alongside of: alphagov/paas-release-ci#58

## How to review

- Code check
- _install dependencies `make dependencies`_
- Run tests `make test` (done with travis)
- Run application `make build && ./bin/rubbernecker`
- Have a look around, change things in pivotal perhaps
